Content.Shared/Anomaly/Effects/Components/TileSpawnAnomalyComponent.cs

```diff
@@ -0,0 +1,79 @@
+using Content.Shared.Maps;
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared.Anomaly.Effects.Components;
+
+[RegisterComponent, Access(typeof(SharedTileAnomalySystem))]
+public sealed partial class TileSpawnAnomalyComponent : Component
+{
+    /// <summary>
+    /// All types of floors spawns with their settings
+    /// </summary>
+    [DataField]
+    public List<TileSpawnSettingsEntry> Entries = new();
+}
+
+[DataDefinition, Serializable, DataRecord]
+public partial record struct TileSpawnSettingsEntry()
+{
```

Comment:
DataRecord makes everything a datafield, also DataDefinition is not required as it duplicates. Serializable also doesn't do anything here.